xdsl/dialects/onnx.py

```diff
@@ -26,6 +26,104 @@
 from xdsl.utils.exceptions import VerifyException
 
 
+def tensor_unidirectional_broadcast_shape(
+    lhs: TensorType[Attribute], rhs: TensorType[Attribute], res: TensorType[Attribute]
+) -> TensorType[Attribute]:
+    """
+    Returns the tensor format of an unidirectional broadcastable shape
+    """
+    lhs_shape = lhs.get_shape()
+    rhs_shape = rhs.get_shape()
+    res_shape = unidirectional_broadcast_shape(list(lhs_shape), list(rhs_shape))
+    res_shape_tensor = cast(TensorType[Attribute], res_shape)
```

Comment:
Doesn't this always fail? `res_shape` should be `None` or `list[int]`, and never a tensor type

Reply:
okay, i delete this and change the return type of the function to `list[int] | None`, but the return type wont be a tensor then right

Reply:
That seems like the right move to me, and then we could use terms like `expected_shape` for whatever is returned from the helper function, and `res_shape` for the shape of the `res` tensor passed in